### PR TITLE
Create email component and coresponding tests

### PR DESCRIPTION
**Pull request by akainic**

### CODE REVIEW

File: _health-care/_js/components/personal-information/VeteranAddressSection.jsx [hunk 1]
```diff
@@ -40,11 +56,16 @@ class VeteranAddressSection extends React.Component {
...
-            <label htmlFor="veteran_email">Email address</label>
-            <input type="text" name="veteran[email]"/>
+            <Email label="Email address"
+                value={this.props.data.email}
+                onValueChange={(update) => {this.props.onStateChange('email', update);}}/>
 
-            <label htmlFor="veteran_email_confirmation">Re-enter Email address</label>
-            <input type="text" name="veteran[email_confirmation]"/>
+            <Email error={
+              this.confirmEmail()
```
Comment (awong-dev) [inline _health-care/_js/components/personal-information/VeteranAddressSection.jsx]: Spacing on this is weird. Move it all onto one line?

File: _health-care/_js/utils/validations.js [hunk 2]
```diff
@@ -17,8 +17,13 @@ function isValidPhone(value) {
   return /^\d{3}-\d{3}-\d{4}$/.test(value);
 }
 
+function isValidEmail(value) {
+  return /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(value);
```
Comment (awong-dev) [inline _health-care/_js/utils/validations.js]: Whoa.  Document where this regexp comes from and why you think it's right?

File: _health-care/_js/components/personal-information/VeteranAddressSection.jsx (hunk 1 above)
Comment (akainic) [inline _health-care/_js/components/personal-information/VeteranAddressSection.jsx]: Yeah, sorry, did that for debugging purposes :)

File: _health-care/_js/utils/validations.js (hunk 2 above)
Comment (akainic) [inline _health-care/_js/utils/validations.js]: Yeah, email addresses are really hard to validate... I found this one on StackOverflow, which I put here in to start with, but not committed to it. I also looked into the validator.js library and their validation logic is even more complex. How specific do we want to be with validating emails? Is this sufficient: some number of characters + @ + some number of characters + . + 2 or more alphabetical characters?

Comment (awong-dev) [inline _health-care/_js/utils/validations.js]: I want a comment citing the stack overflow logic. :)

The point of documentation is a later dev knows why you made the choice. "I just copied from stack overflow" is super useful info cause the other assumption is "the prior dev though about htis hard and probably had some reason for this...let me check things from first principles again."

File: _health-care/_js/components/personal-information/Email.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Input component for collecting a Email number.
+ *
+ * Validates the input data. Does NOT consider "invalid" email numbers such
+ * as 000-000-0000 to be errors. This is to allow integration testing with
+ * fake data.
+ *
+ * Props:
+ * `error` - Function that returns a string. Only necessary for confirming email.
```
Comment (awong-dev) [inline _health-care/_js/components/personal-information/Email.jsx]: ?? This is actually just a string in how you're using it and seems to be exactly like errorMessage in usage. Rename to that?

File: _health-care/_js/components/personal-information/Email.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ * `label` - String. Not required as a prop, can be passed directly to
```
Comment (awong-dev) [inline _health-care/_js/components/personal-information/Email.jsx]: No need to reference ErrorableTextInput int he comment.

To the user of this Component, they shouldn't know that ErrorableTextInput exist. That's an abstraction break.

File: _health-care/_js/components/personal-information/VeteranAddressSection.jsx [hunk 5]
```diff
@@ -1,8 +1,24 @@
...
+  constructor() {
+    super();
+    this.confirmEmail = this.confirmEmail.bind(this);
+  }
+
+  confirmEmail() {
+    let errorMessage = undefined;
+
+    if (this.props.data.email !== this.props.data.emailConfirmation) {
+      errorMessage = 'Please ensure your entries match';
```
Comment (awong-dev) [inline _health-care/_js/components/personal-information/VeteranAddressSection.jsx]: Don't use a variable here.  Just `return 'Please ensure...';` in this branch and `return undefined;` at the bottom. Fewer variables == easier to read.

File: _health-care/_js/utils/validations.js [hunk 6]
```diff
@@ -17,8 +17,14 @@ function isValidPhone(value) {
   return /^\d{3}-\d{3}-\d{4}$/.test(value);
 }
 
+function isValidEmail(value) {
+  // Comes from StackOverflow
```
Comment (awong-dev) [inline _health-care/_js/utils/validations.js]: Include link to the URL?

File: _health-care/_js/components/personal-information/Email.jsx (hunk 3 above)
Comment (awong-dev) [inline _health-care/_js/components/personal-information/Email.jsx]: Also, better to describe the behavior of the API.  For this, if it's set, it is rendered as the error for the input field. That should be the content of this comment.

**Comment by akainic** [comment]
@awong-dev I addressed your comments, PTAL!

**Comment by awong-dev** [comment]
LGTM!

Removed the WIP from your subject. Now merging.

**Pull request merged**